ckanext/stadtzhtheme/commands.py

```diff
@@ -3,177 +3,154 @@
-                if record["alias_of"]:
-                    continue
-
-                logic.check_access("resource_show", context)
-                logic.get_action("resource_show")(context, {"id": record["name"]})
-                print("Resource '%s' found" % record["name"])
-            except logic.NotFound:
-                resource_id_list.append(record["name"])
-                print("Resource '%s' *not* found" % record["name"])
-            except logic.NotAuthorized:
-                print("User is not authorized to perform this action.")
-            except (KeyError, AttributeError) as e:
-                print("Error while handling record %s: %s" % (record, str(e)))
+def _get_datastore_table_page(context, offset=0):
+    # query datastore to get all resources from the _table_metadata
+    result = logic.get_action("datastore_search")(
+        context, {"resource_id": "_table_metadata", "offset": offset}
+    )
+
+    resource_id_list = []
+    for record in result["records"]:
+        try:
+            # ignore 'alias' records
+            if record["alias_of"]:
                 continue
 
-        # are there more records?
-        has_next_page = len(result["records"]) > 0
-
-        return (resource_id_list, has_next_page)
-
-    def _cleanup_storage_files(self, context, resource_path):
-        files_to_delete = []
-        files_to_keep = []
-        for root, dirs, files in os.walk(resource_path, topdown=True):
-            if files:
-                for file in files:
-                    resource_id = "".join(root.split("/")[-2:]) + file
-                    file_path = os.path.join(root, file)
-                    try:
-                        logic.check_access("resource_show", context)
-                        logic.get_action("resource_show")(context, {"id": resource_id})
-                        files_to_keep.append(file_path)
-                    except logic.NotFound:
-                        files_to_delete.append(file_path)
-                    except logic.NotAuthorized:
-                        print("User is not authorized to perform this action.")
-                        sys.exit(1)
-                    except (KeyError, AttributeError) as e:
-                        raise RuntimeError(
-                            "Error while handling record {}: {}".format(
-                                resource_id, str(e)
-                            )
-                        )
-        print("{} files will be deleted:".format(len(files_to_delete)))
-        for file_path in files_to_delete:
-            os.remove(file_path)
-            print("- deleted: {}".format(file_path))
-        return files_to_keep
+            logic.check_access("resource_show", context)
+            logic.get_action("resource_show")(context, {"id": record["name"]})
+            click.echo("Resource '%s' found" % record["name"])
+        except logic.NotFound:
+            resource_id_list.append(record["name"])
```

Comment:
why do we do this in `except` and not in `try`, 
we need to get a list of existing resources hier, right ?